### PR TITLE
KT-24509 Inspection "JUnit tests should return Unit"

### PR DESCRIPTION
**Pull request by t-kameyama**
https://youtrack.jetbrains.com/issue/KT-24509

### CODE REVIEW

File: idea/src/org/jetbrains/kotlin/idea/inspections/MainFunctionReturnUnitInspection.kt [hunk 1]
```diff
@@ -10,39 +10,54 @@ import com.intellij.codeInspection.ProblemDescriptor
...
+            } else {
+                val junitTestFqNames = listOf(FqName("org.junit.Test"), FqName("org.junit.jupiter.api.Test"))
+                if (testAnnotations.none { it.fqName() in junitTestFqNames }) return
+            }
+            if (descriptor.returnType?.let { KotlinBuiltIns.isUnit(it) } == true) return
 
             holder.registerProblem(
                 function.nameIdentifier ?: function,
-                "main should return Unit",
+                "${if (isMain) "main" else "test"} should return Unit",
```
Comment (mglukhikh) [inline idea/src/org/jetbrains/kotlin/idea/inspections/MainFunctionReturnUnitInspection.kt]: I'd suggest to report "JUnit test should return Unit" or "Test function should return Unit" or something like. "test should return Unit" looks not very pretty

File: idea/testData/inspectionsLocal/mainFunctionReturnUnit/junit4Test.kt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+// FIX: Add explicit Unit return type
```
Comment (mglukhikh) [inline idea/testData/inspectionsLocal/mainFunctionReturnUnit/junit4Test.kt]: Please add `// PROBLEM: <problem text>` to one test with JUnit function and one test with main function

**Comment by t-kameyama** [comment]
Fixed

**Comment by mglukhikh** [comment]
Merged

**Pull request closed without merging**